src/analyzers/concept/vehicle-purchase/VehiclePurchaseSolution.ts

```diff
@@ -0,0 +1,213 @@
+import {
+  AstParser,
+  ExtractedFunction,
+  extractExports,
+  extractFunctions,
+  findFirstOfType,
+} from '@exercism/static-analysis'
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/typescript-estree'
+import { readFileSync } from 'fs'
+import path from 'path'
+import { Source } from '~src/analyzers/SourceImpl'
+import { assertPublicApi } from '~src/asserts/assert_public_api'
+import { PublicApi } from '../../PublicApi'
+
+export const NEEDS_LICENSE = 'needsLicense'
+export const CHOOSE_VEHICLE = 'chooseVehicle'
+export const CALCULATE_RESELL_PRICE = 'calculateResellPrice'
+
+/**
+ * Verify that `needsLicense` does not include an unnecessary if-statement where
+ * the student returns `true`/`false`.
+ */
+class NeedsLicense extends PublicApi {
+  constructor(public readonly implementation: ExtractedFunction) {
+    super(implementation)
+  }
+
+  public get isOptimal(): boolean {
+    if (this.implementation.params.length !== 1) {
+      return false
+    }
+
+    let foundSuboptimalNode = false
+
+    this.traverse({
+      enter() {
+        foundSuboptimalNode = true
+      },
+
+      [AST_NODE_TYPES.ReturnStatement]() {
+        foundSuboptimalNode = false
+      },
+
+      [AST_NODE_TYPES.BlockStatement]() {
+        foundSuboptimalNode = false
+      },
```

Comment:
This says it optimally looks like this:

```javascript
function x(first) {
  return
}
```

...assuming it's not yet finished?

Reply:
Yeah not finished yet. Still trying to think about the process to take in the traversal process.